libraries/core/src/main/java/com/paritytrading/philadelphia/FIXConnection.java

```diff
@@ -133,7 +133,7 @@ public FIXConnection(ReadableByteChannel inboundChannel, GatheringByteChannel ou
         this.senderCompId = config.getSenderCompID();
         this.targetCompId = config.getTargetCompID();
 
-        this.parser = new FIXMessageParser(config, new MessageHandler(listener));
+        this.parser = new FIXMessageParser(config, this, new MessageHandler(listener));
```

Comment:
`MessageHandler` being a private inner class is an issue. In particular, at some point, I needed to customize some of the handling and I could not subclass and override. I used ByteBuddy to intercept the method, and later removed the feature as I found a way ton achieve the same in different part of the application.

My view is that we should inline the message parser in the connection class (using protected methods, so people can customize), and pass the message handler from outside, providing default implementation on `null`.

Reply:
I agree on the `MessageHandler` issue. The plan is to move it out from `FIXConnection` and make it possible for the users of the library to provide their own replacement if they wish. Currently `MessageHandler` implements a subset of the FIX protocol, but this subset is currently too little for some applications and unnecessarily much for others.